selendroid-standalone/src/main/java/io/selendroid/standalone/server/grid/SelfRegisteringRemote.java

```diff
@@ -156,4 +166,40 @@ private JSONObject getConfiguration() throws JSONException {
     configuration.put("remoteHost", "http://" + config.getServerHost() + ":" + config.getPort());
     return configuration;
   }
+
+  private boolean isAlreadyRegistered() {
+    HttpClient client = HttpClientUtil.getHttpClient();
+
+    try {
+      String tmp = config.getRegistrationUrl().replace("/grid/register", "/grid/api/proxy"); //TODO find a better way
```

Comment:
Well, this is kinda for discussion. 
I've left this TODO here as currently there is no better way to do it. 
See, that selendroid's only knowledge of hub is **'-hub'** argument. Also, see in [SelendroidConfiguration.java](https://github.com/selendroid/selendroid/blob/master/selendroid-standalone/src/main/java/io/selendroid/standalone/SelendroidConfiguration.java) that a argument **-hub** which is actually not a hub, but a URL for registering (i.e. it is not a host name or port, but a string with URL - http://localhost:4444/grid/register or something). 
The best, if it would be hub-host and hub-port, as originally in grid sources, then there would no be this TODO. But I guess it's too late to change this.

At this point, perhaps a more correct way (not faster, but more correct), would be:
1. Take string value of **'-hub'** argument
2. Create a URL object of it
3. Take Scheme, host and port parts from URL object
4. Form a new URL with path **'/grid/api/proxy'**

If that looks like an OK refactoring, I'll apply the change.